src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java

```diff
@@ -261,96 +262,184 @@ public ExamScoresDTO calculateExamScores(Long examId) {
-                            exercise.getMaxPoints(), relevantResult.getScore(), achievedPoints, hasNonEmptySubmission));
+                    .filter(studentParticipation -> studentParticipation.getStudent().get().getId().equals(studentExam.getUser().getId())).toList();
+            ExamScoresDTO.StudentResult studentResult = calculateStudentResultWithGrade(studentExam, participationsOfStudent, exam, scores, objectMapper, gradingScale, true);
+            scores.studentResults.add(studentResult);
+        }
+
+        // Updating exam information in DTO
+        int numberOfStudentResults = scores.studentResults.size();
+        if (numberOfStudentResults != 0) {
+            double sumOverallPoints = scores.studentResults.stream().mapToDouble(studentResult -> studentResult.overallPointsAchieved).sum();
+            scores.averagePointsAchieved = sumOverallPoints / numberOfStudentResults;
+        }
+
+        return scores;
+    }
+
+    /**
+     * Calculates max points, max bonus points and achieved points per exercise if the given studentExam is assessed.
+     * Includes the corresponding grade and grade type as well if a GradingScale is set for the relevant exam.
+     * @param studentExam a StudentExam instance that will have its points and grades calculated if it is assessed
+     * @param participationsOfStudent StudentParticipation list for the given studentExam
+     * @return Student Exam results with exam grade calculated if applicable
+     */
+    @NotNull
+    public StudentExamWithGradeDTO calculateStudentResultWithGradeAndPoints(StudentExam studentExam, List<StudentParticipation> participationsOfStudent) {
+        Exam exam = studentExam.getExam();
+
+        // Adding exam information to DTO
+        var scores = new ExamScoresDTO(exam.getId(), exam.getTitle(), exam.getMaxPoints());
+        var objectMapper = new ObjectMapper();
+
+        Optional<GradingScale> gradingScale = gradingScaleRepository.findByExamId(exam.getId());
+        ExamScoresDTO.StudentResult studentResult = calculateStudentResultWithGrade(studentExam, participationsOfStudent, exam, scores, objectMapper, gradingScale, false);
+        var studentExamWithGradeDTO = new StudentExamWithGradeDTO(studentExam, studentResult);
+
+        gradingScale.ifPresent(scale -> studentExamWithGradeDTO.gradeType = scale.getGradeType());
+
+        List<Exercise> exercises = studentExam.getExercises();
+        studentExamWithGradeDTO.maxPoints = calculateMaxPointsSum(exercises, exam.getCourse());
+        studentExamWithGradeDTO.maxBonusPoints = calculateMaxBonusPointsSum(exercises, exam.getCourse());
+        studentExamWithGradeDTO.achievedPointsPerExercise = calculateAchievedPointsForExercises(participationsOfStudent, exam.getCourse());
+
+        return studentExamWithGradeDTO;
+    }
+
+    @NotNull
+    private ExamScoresDTO.StudentResult calculateStudentResultWithGrade(StudentExam studentExam, List<StudentParticipation> participationsOfStudent, Exam exam,
```

Comment:
Considering the complex and non-trivial nature of this method I think it should also have a JavaDoc comment describing its function.

Reply:
I added Javadoc for this.